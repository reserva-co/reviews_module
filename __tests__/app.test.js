/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/extensions */
/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/components/app.jsx';

const $ = require('jquery');

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('ReviewDiv');
  ReactDOM.render(<App />, div);
});

it('should render correctly with no props', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});

it('calls `componentDidMount` before rendering', () => {
  const onDidMount = jest.fn();
  App.prototype.componentDidMount = onDidMount;
  mount(<App />);
  expect(onDidMount).toHaveBeenCalled();
});

it('getTotalStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getTotalStars();
  expect(wrapper.state('totalScore')).toBe(NaN);
});

it('getAccuracyStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getAccuracyStars();
  expect(wrapper.state('accuracy')).toBe(NaN);
});

it('getCommunicationStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getCommunicationStars();
  expect(wrapper.state('communication')).toBe(NaN);
});

it('getCleanlinessStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getCleanlinessStars();
  expect(wrapper.state('cleanliness')).toBe(NaN);
});

it('getLocationStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getLocationStars();
  expect(wrapper.state('location')).toBe(NaN);
});

it('getCheckInStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getCheckInStars();
  expect(wrapper.state('checkIn')).toBe(NaN);
});

it('getValueStars sets score in state', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.getValueStars();
  expect(wrapper.state('value')).toBe(NaN);
});

jest.mock('jquery');

beforeEach(() => jest.resetModules());

it('should return a review', () => $.get('http://localhost3000/api/reviews/1', (data) => {
  expect(data).toBeDefined();
  expect(data.id).toEqual(1);
}));
