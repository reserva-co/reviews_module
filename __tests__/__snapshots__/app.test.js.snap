// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<styled.div>
  <styled.div
    style={
      Object {
        "borderBottomColor": "#e4e4e4",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
      }
    }
  >
    <styled.h2>
      0 Reviews
    </styled.h2>
    <div
      style={
        Object {
          "marginLeft": "15px",
          "marginTop": "25px",
        }
      }
    >
      <StarRatings
        changeRating={null}
        className="main-star"
        gradientPathName=""
        ignoreInlineStyles={false}
        name="rating"
        numberOfStars={5}
        rating={0}
        starDimension="20px"
        starEmptyColor="rgb(203, 211, 227)"
        starHoverColor="rgb(230, 67, 47)"
        starRatedColor="teal"
        starSpacing="7px"
        svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        svgIconViewBox="0 0 51 48"
        typeOfWidget="Star"
      />
    </div>
    <Search
      handleSearch={[Function]}
    />
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.h3>
          Accuracy
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
      <styled.div>
        <styled.h3>
          Communication
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
      <styled.div>
        <styled.h3>
          Cleanliness
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h3>
          Location
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
      <styled.div>
        <styled.h3>
          Check-In
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
      <styled.div>
        <styled.h3>
          Value
        </styled.h3>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "15px",
              "marginTop": "10px",
            }
          }
        >
          <StarRatings
            changeRating={null}
            className="secondary-star"
            gradientPathName=""
            ignoreInlineStyles={false}
            name="rating"
            numberOfStars={5}
            rating={0}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="teal"
            starSpacing="7px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
      </styled.div>
    </styled.div>
  </styled.div>
  <NoResults />
</styled.div>
`;
